core/embed/boardloader/main.c

```diff
@@ -167,9 +167,19 @@ static uint32_t check_sdcard(void) {
     _Static_assert(IMAGE_CHUNK_SIZE >= BOOTLOADER_IMAGE_MAXSIZE,
                    "BOOTLOADER IMAGE MAXSIZE too large for IMAGE_CHUNK_SIZE");
 
-    if (sectrue != (check_single_hash(
-                       hdr->hashes, ((const uint8_t *)sdcard_buf) + hdr->hdrlen,
-                       hdr->codelen))) {
+    const uint32_t headers_end_offset = hdr->hdrlen;
+    const uint32_t code_start_offset = IMAGE_CODE_ALIGN(headers_end_offset);
+
+    for (uint32_t i = headers_end_offset; i < code_start_offset; i++) {
+      if (sdcard_buf[i] != 0) {
```

Comment:
`headers_end_offset` and `code_start_offset` are both offsets in bytes, but `sdcard_buf` is defined as an array of `uint32_t`.

Reply:
fixed, thanks

https://github.com/trezor/trezor-firmware/pull/4090/commits/7c61e651fccbce7cdded30cb910c85d756ab5b33